### PR TITLE
RFP: Mergable Receipts Delivery

### PR DESCRIPTION
**Pull request by plotozhu**
Mergable Receipts Delivery, Algorithm to cut down bandwidth and tps for retrieval market

### CODE REVIEW

File: open-grant-proposals/open-proposal-mrd.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+**Proposer:** [@plotozhu](https://github.com/plotozhu)
+
+**Do you agree to open source all work you do on behalf of this RFP and dual-license under MIT and APACHE2 licenses?:**
+"YES"
+
+# Project Description
+For retrieve markets of Filecoin, data traffic occurs off chain and results should be submitted to chain. To achieve this, client node should send receipts to retrieving miner on accepted data.
+
+Cause it was designed as every 256KB per chunk and length of receipt should be about 160Bytes, occupying about 0.1% bandwith of total nodes. It would mean that if filecoin had 100T bandwith,  on chain bandwith for receipts  would be about 100G, and tps would reach 100G/160 = 625M, which was unaccpetable.
```
Comment (pooja) [inline open-grant-proposals/open-proposal-mrd.md]: Hi @plotozhu could you clarify where you got the 256KB number from?

File: open-grant-proposals/open-proposal-mrd.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+We present a algorithm call MRD(Mergable Receipts Delivery) to solve this problem.
+
+
+## Value
+With MRD, client node sends receipt to retrieval node on each data chunk, retrieval node collects receipts. Multiple receipts can be merged to one, retrieval miner can merge, pack these receipts and broadcast to chain periodly.
+We have tested this algorithm in thounds of retrieval nodes and millions of clients, and proved it worked well.
```
Comment (pooja) [inline open-grant-proposals/open-proposal-mrd.md]: Have you tested this algorithm with thousands of **Filecoin** retrieval nodes? Please say more

File: open-grant-proposals/open-proposal-mrd.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+| 1 | algrorithm description and proof of mrd | $50K | 4 weeks |
+| 2 | prototype source code of mrd  | $100K | 6 weeks |
+| 3 | Completed implementation of mrd | $200K | 6 weeks |
+
+Can be processed parallelly
+| Milestone No. | Milestone Description | Funding | Estimated Timeframe |
+| --- | --- | --- | --- |
+| 1 | algrorithm description and proof of mrd | $50K | 4 weeks |
+| 2 | prototype source code of mrd  | $50K | 6 weeks |
+| 3 | Completed implementation of mrd | $100K | 6 weeks |
```
Comment (pooja) [inline open-grant-proposals/open-proposal-mrd.md]: Can you clarify what the difference is between the first budget estimate and this one?

File: open-grant-proposals/open-proposal-mrd.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+## Total Budget Requested
+
+Total budget costs from $200K up to $550K, to achieve a robust and stable version to used in production version
```
Comment (pooja) [inline open-grant-proposals/open-proposal-mrd.md]: Can you clarify why the total budget is so high and the range is so large?

Comment (plotozhu) [reply open-grant-proposals/open-proposal-mrd.md]: Sorry for errors in budget describing.  There are budget for MRD and DDPFT each.  They can be processed in parallel:
This is for MRD
| Milestone No. | Milestone Description | Funding | Estimated Timeframe |
| --- | --- | --- | --- |
| 1 | algrorithm description and proof of MRD | $50K | 4 weeks |
| 2 | prototype source code of MRD  | $50K | 6 weeks |
| 3 | Completed implementation of MRD | $100K | 6 weeks |

This is for DDPFT:
| Milestone No. | Milestone Description | Funding | Estimated Timeframe |
| --- | --- | --- | --- |
| 1 | algrorithm description and proof of DDPFT | $50K | 4 weeks |
| 2 | prototype source code of DDPFT  | $100K | 6 weeks |
| 3 | Completed implementation of DDPFT | $200K | 6 weeks |

**Comment by pooja** [comment]
We've finished the wave 1 grant selection process. Can you please email devgrants@filecoin.org so we can share our decision and next steps, if any? In your email, please include a link to your proposal PR so we know who we're talking to!

Thanks,
Filecoin DevGrants team

**Comment by pooja** [comment]
Greetings,

Thank you for your application to the Filecoin DevGrants program. We appreciate your proposal!

You were not selected for a grant in Wave 1, but we hope that you consider applying again for a future DevGrant in a later wave.

We selected grants based on:
- what makes sense to build now based on where the Filecoin node implementations are
- the likelihood of your team being able to work independently of the Filecoin engineering teams working on the core client implementations
- degree of refactoring your proposal may require as the software clients evolve
- the integrity of the proposal and near-term value to the Filecoin ecosystem
- your commitment to the ecosystem and desire to maintain your project longer-term

As the Filecoin implementations advance towards Testnet and Mainnet, we hope you continue to consider new proposals for projects. We would love to see more applications from you in the future.

Please stay tuned for more Filecoin grant RFPs. :)

**Pull request closed without merging**